src/WebdriverClassicDriver.php

```
@@ -1,21 +1,1187 @@
+                    return $element->isSelected() ? $element->getAttribute('value') : null;
+
+                case $widgetType === 'select':
+                    $selectElement = new WebDriverSelect($element);
+                    $selectedOptions = array_map(
+                        static fn(WebDriverElement $option) => $option->getAttribute('value'),
+                        $selectElement->getAllSelectedOptions()
+                    );
+                    return $selectElement->isMultiple() ? $selectedOptions : ($selectedOptions[0] ?? '');
+
+                default:
+                    return $this->getWebDriver()->isW3cCompliant()
+                        ? $element->getDomProperty('value')
+                        : $this->executeJsOnElement($element, 'return arguments[0].value');
+            }
+        } catch (Throwable $e) {
+            throw new DriverException("Cannot retrieve $widgetType value: {$e->getMessage()}", 0, $e);
+        }
+    }
+
+    public function setValue(
+        #[Language('XPath')]
+        string $xpath,
+        $value
+    ): void {
+        $element = $this->findElement($xpath);
+        $elementName = strtolower($element->getTagName() ?? '');
+
+        switch ($elementName) {
+            case 'textarea':
+                if (!is_string($value)) {
+                    throw new DriverException('Textarea value must be a string');
+                }
+                $element->clear();
+                $element->sendKeys($value);
+                break;
+
+            case 'select':
+                if (is_array($value)) {
+                    $this->deselectAllOptions($element);
+                    foreach ($value as $option) {
+                        $this->selectOptionOnElement($element, $option, true);
+                    }
+                    return;
+                }
+                $this->selectOptionOnElement($element, (string)$value);
```

This should throw an exception for boolean values instead of casting them to string.